File: src/SIStorage.Service.Contract/Requests/PackageSortMode.cs
namespace SIStorage.Service.Contract.Requests;

/// <summary>
/// Defines a packages sort mode.
/// </summary>
public enum PackageSortMode
{
    /// <summary>
    /// Sort by name.
    /// </summary>
    Name = 0,

    /// <summary>
    /// Sort by created date.
    /// </summary>
    CreatedDate = 1,

    /// <summary>
    /// Sort by download count.
    /// </summary>
    DownloadCount = 2,

    /// <summary>
    /// Sort by rating.
    /// </summary>
    Rating = 3,
}

File: web/src/models/PackageSortMode.ts
/** Defines a packages sort mode. */
const enum PackageSortMode {
	/** Sort by name. */
	Name = 0,

	/** Sort by created date. */
	CreatedDate = 1,

	/** Sort by download count. */
    DownloadCount = 2,

	/** Sort by rating. */
    Rating = 3,
}

export default PackageSortMode;